<?xml version="1.0"?>

<launch>

  <arg name="display_results_with_image_view" default="true"/>   <!-- set to false if you do not like to display the camera image with names attached to detected faces within an image_view window -->
  <arg name="face_positions_topic_name_in" value="face_detector/face_positions"/>
  <arg name="in_topic_image_name" value="sensor_message_gateway/colorimage_out"/>

  <!-- display node for the detections -->
  <node pkg="cob_people_detection" type="people_detection_display_node" name="people_detection_display" output="screen">
    <rosparam command="load" file="$(find cob_people_detection)/ros/launch/people_detection_display_params.yaml"/>
    <remap from="~face_position_array" to="detection_tracker/face_position_array"/>
    <remap from="~face_detections" to="$(arg face_positions_topic_name_in)"/>
    <remap from="~colorimage_in" to="$(arg in_topic_image_name)"/>
  </node>

  <!-- Image viewer display -->
  <node if="$(arg display_results_with_image_view)" name="people_detection_viewer" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="people_detection_display/face_position_image"/>
    <param name="autosize" value="true"/>
  </node>

</launch>
